### ISSUE TITLE
3.x Variables - Substitute in Files can not locate Octopus.Platform.dll

### ISSUE THREAD

**Issue by aarondandy** [OctopusDeploy/Issues#2051]
After upgrading to 3.x our Variables - Substitute in Files steps no longer work. This appears to be due to the 'Octopus.Tentacle.Agent.ProgramDirectoryPath' variable pointing to a directory (in our case `D:` ) that does not contain "Octopus.Platform.dll".

**Comment by alfhenrik** [comment]
Hi @aarondandy,

That specific step template is not compatible with Octopus 3.x as it is using a 2.6 specific DLL (Octopus.Platform.dll was removed as part of 3.0), and all that functionality has been moved into its own package.

I've raised OctopusDeploy/Library#223 to create a step template that does the same thing and is compatible with 3.x.

One possible workaround is to use the `Substitute variables in Files` feature of a `NuGet package` step but I'm not sure this will work for your situation (my guess is that it probably won't as you're using the step template in the first place).

Sorry for the inconvenience caused by this issue.

I am closing this issue, and any further communication will be done in the above mentioned GH issue.

Thank you and kind regards,
Henrik
